sdk/python/feast/feature_store.py

```diff
@@ -666,14 +666,16 @@ def apply(
             ob for ob in objects_to_delete if isinstance(ob, FeatureView)
         ]
 
-        self._get_provider().update_infra(
-            project=self.project,
-            tables_to_delete=views_to_delete if not partial else [],
-            tables_to_keep=views_to_update,
-            entities_to_delete=entities_to_delete if not partial else [],
-            entities_to_keep=entities_to_update,
-            partial=partial,
-        )
+        # Don't call update if there isn't an online store
+        if not isinstance(self.config.online_store, Dict):
```

Comment:
Provider is responsible for other things, not only online store (eg, AWS Provider deploys lambda function), so we shouldn't skip `update_infra`